unitex/src/fr/umlv/unitex/leximir/delac/ChooseDelac.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Unitex
+ *
+ * Copyright (C) 2001-2018 Université Paris-Est Marne-la-Vallée <unitex@univ-mlv.fr>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA.
+ *
+ */
+package fr.umlv.unitex.leximir.delac;
+
+import javax.swing.JFrame;
+
+import fr.umlv.unitex.common.project.manager.GlobalProjectManager;
+import fr.umlv.unitex.frames.UnitexInternalFrameManager;
+import java.io.File;
+import javax.swing.JFileChooser;
+import javax.swing.JOptionPane;
+import javax.swing.filechooser.FileNameExtensionFilter;
+import fr.umlv.unitex.leximir.model.DictionaryPath;
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * @author Rojo Rabelisoa
+ * @author Anas Ait cheikh
+ */
+public class ChooseDelac extends javax.swing.JInternalFrame {
+
+    /**
+     * Creates new form ChooseDelac
+     */
+    File lastLink;
+
+    public ChooseDelac() {
+        super("Edit Delac", true, true, true, true);
+        initComponents();
+        jRadioallDelac.setSelected(true);
+        buttonGroup1.add(jRadioallDelac);
+        buttonGroup1.add(jRadioBrowse);
+        this.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+        lastLink = new File(DictionaryPath.allDela + "/tmp_delac.txt");
+        if (lastLink.exists()) {
+            try {
+            	try( BufferedReader br = new BufferedReader(new FileReader(lastLink)) ) {
+            		jTextField1.setText(br.readLine());
```

Comment:
Use instead [`UnicodeIO.readLine`](../../tree/master/unitex/src/fr/umlv/unitex/io/UnicodeIO.java). Quoting Sebastien:

```
Only use unicode functions that are in the fr.umlv.unitex.io.UnicodeIO 
class. This is very important, since IO functions do not have the same 
behavior on little-endian and big-endian systems. If you miss a function,
please code it from the primitive functions of this library. 
```

Reply:
Okay